[Security] Entity relationships in Entity Analytics (#6850)
elastic/docs-content: solutions/security/advanced-entity-analytics/entity-relationships.md

---
navigation_title: Entity relationships
description: Learn how the entity store derives entity relationships such as access patterns and communication links from your ingested integration data.
applies_to:
  stack: ga 9.4+
  serverless:
    security: ga
products:
  - id: security
  - id: cloud-serverless
---

# Entity relationships [entity-relationships]

Entity relationships represent the connections between entities — for example, `user-1 communicates_with host-1`. The [entity store](/solutions/security/advanced-entity-analytics/entity-store.md) stores these connections as a fixed set of relationship keys on each entity record, under the schema `entity.relationships.<key>.ids`. Relationships feed entity analytics views, including the [entity graph](/solutions/security/advanced-entity-analytics/view-entity-details.md#visualizations) in the entity details flyout. Relationships are populated only when the entity store is enabled and populated in the active space.

> The relationship model is the foundation for representing how entities interact across your environment. A clear mapping between integrations, fields, and relationship keys helps you understand:

Reviewer: ```suggestion
The relationship model represents how entities interact across your environment. Integrations, fields, and relationship keys help you understand:
```
Shorter option, I think it still intros the bullets well.

* Which fields a third-party integration must emit for its data to participate in a given relationship.

> * Which relationships exist and how they are populated.

Reviewer: Wydt of switching the order of these bullets? To me, the second one seems like the primary thing to highlight

## How relationships are produced [entity-relationships-maintainers]

> Entity relationships are produced by *maintainers* — background tasks that scan ingested events from supported integrations and write the result into the entity store.

Reviewer: ```suggestion
Entity relationships are produced by *maintainers* — background tasks that detect relationships in ingested events from supported integrations and record them in the entity store.
```

> Each maintainer scans a specific integration's index pattern for events that match an {{esql}} filter, identifies an actor (a user) and a target (a user, host, or service), computes the target's entity unique identifier (EUID), and writes the relationship onto the actor entity record under `entity.relationships.<key>.ids`.

Reviewer: ```suggestion
Each maintainer scans a specific integration's index pattern for events that match an {{esql}} filter, identifies an actor (a user) and a target (a user, host, or service), computes the target entity's unique identifier (EUID), and adds the relationship to the actor entity as `entity.relationships.<key>.ids`.
```

Reviewer: Tried to simplify. Might have gone too far with removing "record", if "entity record" is a concept

## Requirements for an event to participate [entity-relationships-requirements]

For an event to be picked up by a maintainer, it must meet the following conditions:

* Land in the integration's index pattern (`logs-<integration>-<namespace>`).
* Carry the actor identity field used by the maintainer (typically `user.name`, `user.email`, or an integration-specific principal field).
* Match the maintainer's {{esql}} `WHERE` clause for that integration.
* Carry a resolvable target identity (for example, `host.id`, `host.target.entity.id`, or `user.target.email`).

## Supported relationships [entity-relationships-supported]

The following sections describe each supported relationship key, the integrations that populate it, and the criteria an event must match. To populate a relationship, ingest events into one of the listed integration indices with the required actor and target identity fields.

> ### `accesses_frequently` / `accesses_infrequently` [entity-relationships-accesses]

Reviewer: What about the making this title and the following a little more descriptive, like "The 'accesses_infrequently' / 'accesses_frequently' entity relationships"

Might add some more context for SEO.

> These relationships are bucketed by event count over a 30-day window. A `COUNT(*) >= 4` results in `accesses_frequently`; otherwise, the relationship is `accesses_infrequently`.

Reviewer: How does the time window work? e.g. when does the time window start. Is it always the last 30 days? If so, what about 
```suggestion
These relationships are bucketed by their event count for the last 30 days. A `COUNT(*) >= 4` results in `accesses_frequently`; otherwise, the relationship is `accesses_infrequently`.
```

Author: Hey @alexreal1314, could you help clarify this? 🙏

Reviewer 2: @natasha-moore-elastic yes the maintainers query events from the last 30 days, while the maintainers run it on a daily cadence.

:::{table}
:widths: 2-3-1-6

| Integration | {{ipm-cap}} | Target type | {{esql}} filter applied |
| --- | --- | --- | --- |
| {{elastic-defend}} | `logs-endpoint.events.security-<ns>` | host | `event.action == "log_on" AND process.Ext.session_info.logon_type IN ("RemoteInteractive","Interactive","Network") AND event.outcome == "success"` |
| {{aws}} CloudTrail | `logs-aws.cloudtrail-<ns>` | host | `event.module == "aws" AND event.action IN ("StartSession","SendSSHPublicKey") AND event.outcome == "success"` |
| System Auth | `logs-system.auth-<ns>` | host | `event.category IN ("authentication","session") AND event.action == "ssh_login" AND event.outcome == "success"` |
| System Security | `logs-system.security-<ns>` | host | `event.action IN ("logged-in","logged-in-explicit") AND event.code IN ("4624","4648") AND winlog.logon.type IN ("Interactive","RemoteInteractive","CachedInteractive") AND event.outcome == "success"` (excludes built-in system accounts) |

:::

> To populate `entity.relationships.accesses_frequently.ids` or `entity.relationships.accesses_infrequently.ids` for a user, ingest events into one of the supported integration indices with the actor identity (`user.name` or `user.id`) and a target host identity (`host.id` or `host.name`), and ensure they match the {{esql}} filter.

Reviewer 2: In addition an actor fields could also be `user.email`. similarly for the target it should be one of `host.id`, `host.name`, `host.hostname`.
